Lib/BitUtils.hpp

```diff
@@ -70,6 +70,45 @@ class BitUtils
     return bits;
 #endif
   }
+
+  // compute a 64-bit mask starting at `lower` and ending just before `upper`
+  template<unsigned lower, unsigned upper>
+  static constexpr uint64_t bitmask64() {
+      static_assert(lower < upper, "empty range");
+      static_assert(upper - lower <= 64, "too many bits");
+      uint64_t mask = ~0;
+      mask >>= lower;
+      mask <<= lower;
+      mask <<= 64 - upper;
+      mask >>= 64 - upper;
+      return mask;
+  }
+
+  // get the bits of `_content` between `lower` and `upper`
+  template<unsigned lower, unsigned upper, class T>
+  static uint64_t getBits(const T& t) {
+    auto mask = bitmask64<lower, upper>();
+    return (t._content & mask) >> lower;
+  }
+
+  // set the bits of `_content` between `lower` and `upper` to corresponding bits of `data`
+  template<unsigned lower, unsigned upper, class T>
+  static void setBits(T& t, uint64_t data) {
```

Comment:
I guess these should be called `getBits64` and `setBits64`, but I'll fix it eventually.